code/xcvm/cosmwasm/contracts/gateway/src/interpreter.rs

```diff
@@ -29,9 +30,15 @@ pub(crate) fn force_instantiate(
 	let call_origin = CallOrigin::Local { user: user_origin };
 	let interpreter_origin = InterpreterOrigin {
 		user_origin: call_origin.user(config.network_id),
-		salt: b"default".to_vec(),
+		salt: salt.clone().map(|x| x.into_bytes()).unwrap_or_default(),
```

Comment:
How about

```
let salt = salt.map_or(Vec::new(), |salt| salt.into_bytes());
```

at the beginning of the function and then you don’t have to repeat
this map operation in two places (on line 33 and 40).

Reply:
fixed in current PR i am on.